domain/containerimageresourcestore/service/service.go

```diff
@@ -121,3 +127,32 @@ func (s Service) Remove(
 	}
 	return nil
 }
+
+// getSizeAndFingerprint gets the size and fingerprint that a DockerImageDetails
+// struct will have when exported as a blob from the container image metadata
+// service.
+func getSizeAndFingerprint(dockerDetails docker.DockerImageDetails) (int64, store.Fingerprint, error) {
+	data, err := json.Marshal(dockerDetails)
+	if err != nil {
+		return 0, store.Fingerprint{}, errors.Errorf("marshelling container image metadata: %w", err)
+	}
+	reader := bytes.NewReader(data)
+	length := int64(reader.Len())
```

Comment:
todo: Add a comment on this change to remind us of why we read the Len first.

Reply:
Added.